src/BoilerPlate/Entity.cpp

```diff
@@ -29,35 +31,68 @@ void Entity::DrawEntity()
 
 void Entity::OnWindowResize(float newWindowHeight, float newWindowWidth)
 {
-	minWindowHeight = -newWindowHeight / 2;
-	maxWindowHeight = -minWindowHeight;
-	minWindowWidth = -newWindowWidth / 2;
-	maxWindowWidth = -minWindowWidth;
+	m_minWindowHeight = -newWindowHeight / 2;
+	m_maxWindowHeight = -m_minWindowHeight;
+	m_minWindowWidth = -newWindowWidth / 2;
+	m_maxWindowWidth = -m_minWindowWidth;
 }
 
 void Entity::ArrangeEntityPoints(void)
 {
 
 }
 
-void Entity::Update()
+void Entity::Update(float deltaTime)
 {
+	m_entityPosition.x += m_entityVelocity.x * static_cast<float> (deltaTime); // displacement = velocity * time
+	m_entityPosition.y += m_entityVelocity.y * static_cast<float> (deltaTime);
 
+	m_entityPosition.x = Warp(m_entityPosition.x, m_minWindowWidth, m_maxWindowWidth);
+	m_entityPosition.y = Warp(m_entityPosition.y, m_minWindowHeight, m_maxWindowHeight);
 }
 
 void Entity::Render(void)
 {
 
 }
 
-void Entity::MoveForward(void)
-{
-
-}
-
 float Entity::Warp(float coordinate, float min, float max)
 {
 	if (coordinate < min) return max - (min - coordinate);
 	if (coordinate > max) return min + (coordinate - max);
 	return coordinate;
-}
\ No newline at end of file
+}
+
+void Entity::ApplyImpulse(Vector2 impulse)
+{
+	if (m_entityMass > 0)
+	{
+		m_entityVelocity.x -= (impulse.x / m_entityMass) * sinf(m_utility.ToRadians(m_entityOrientation));
+		m_entityVelocity.y += (impulse.y / m_entityMass) * cosf(m_utility.ToRadians(m_entityOrientation));
+	}
+}
+
+void Entity::DrawBoundingCircle(void)
+{
+	if (m_showingCircles) 
+	{
+		int lineAmount = 100;
+		float twicePI = 2.0f * m_utility.PI;
+
+		glLoadIdentity();
```

Comment:
Add a comment describing why the `glLoadIdentity` call is needed. Just for personal reference.